src/ucp/wireup/wireup.c

```diff
@@ -1242,12 +1278,14 @@ static void ucp_wireup_msg_dump(ucp_worker_h worker, uct_am_trace_type_t type,
     ucs_free(unpacked_address.address_list);
 }
 
-int ucp_worker_iface_is_tl_p2p(const uct_iface_attr_t *iface_attr)
+int ucp_worker_iface_is_tl_p2p(const uct_iface_attr_t *iface_attr,
+                               int prefer_p2p)
 {
     uint64_t flags = iface_attr->cap.flags;
+    int is_p2p     = !!(flags & UCT_IFACE_FLAG_CONNECT_TO_EP);
 
-    return (flags & UCT_IFACE_FLAG_CONNECT_TO_EP) &&
-           !(flags & UCT_IFACE_FLAG_CONNECT_TO_IFACE);
+    return prefer_p2p ? is_p2p :
```

Comment:
seems that if you use a parameter (perfer_p2p) to decide if the iface is p2p or not then the name of the function doesn't match. maybe ucp_worker_iface_is_prefer_tl_p2p and it could use the original ucp_worker_iface_is_tl_p2p?

Reply:
IMO this doesn't match better. here, `prefer_p2p` works as a hint if both caps are supported. original `ucp_worker_iface_is_tl_p2p` can't be used in this case

Reply:
ucp_worker_iface_is_tl_p2p questions if the iface supports p2p or not. if you make it decide which one to use, then you are not questioning a capability but rather forcing an action.
it's just naming so it's not critical but imo would be clearer to rename.

Reply:
need to fix this, see comment on ucp_worker_is_tl_p2p